### PR TITLE
[Calendar] Update recursive meeting

### PR DESCRIPTION
**Pull request by zidaneymar**
## Description

Add support for updating recursive meetings. If "daily", "weekly" or "monthly" are recognized, the event id will be replaced by SeriesMasterId(Outlook) or  RecurringEventId(Google), and a series of repeating event will be modified at the same time.

## Testing Steps

Steps to change recurring meeting:
image

Steps to change a single instance of recurring meetings:
image

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/UpdateEvent/UpdateEventDialog.cs [hunk 1]
```diff
@@ -137,6 +138,18 @@ public async Task<DialogTurnResult> UpdateEventTime(WaterfallStepContext sc, Can
                     updateEvent.TimeZone = TimeZoneInfo.Utc;
                     updateEvent.Id = origin.Id;
 
+                    if (!string.IsNullOrEmpty(state.RecursiveToken))
+                    {
+                        if (origin.Source == EventSource.Microsoft)
```
Comment (Theodosia1005) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/UpdateEvent/UpdateEventDialog.cs]: Please add a new Id like SerriesId in EventModel, and move this logic to EventModel.
change here to if !string.IsNullOrEmpty(state.RecursiveToken origin.Id = origin.SerriesId

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/UpdateEvent/UpdateEventDialog.cs [hunk 2]
```diff
@@ -137,6 +138,18 @@ public async Task<DialogTurnResult> UpdateEventTime(WaterfallStepContext sc, Can
                     updateEvent.TimeZone = TimeZoneInfo.Utc;
                     updateEvent.Id = origin.Id;
 
+                    if (!string.IsNullOrEmpty(state.RecursiveToken))
```
Comment (KayMKM) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/UpdateEvent/UpdateEventDialog.cs]: how about the case that the found meeting is not a weekly meeting?

**Comment by lzc850612** [comment]
I believe you meant Recurring Meeting :)

You don't wanna get stuck in a recursive meeting ;)

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/CalendarSkillDialog.cs [hunk 3]
```diff
@@ -20,6 +20,7 @@
 using Newtonsoft.Json.Linq;
 using static Microsoft.Recognizers.Text.Culture;
 using Microsoft.Bot.Builder.AI.Luis;
+using CalendarSkill.Dialogs.Shared.Resources.Strings;
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/CalendarSkillDialog.cs]: I think the order of the using statements here is probably messed up.

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/Resources/Strings/CalendarCommonStrings.resx [hunk 4]
```diff
@@ -120,6 +120,15 @@
   <data name="AllDay" xml:space="preserve">
     <value>All day</value>
   </data>
+  <data name="DailyToken" xml:space="preserve">
+    <value>daily</value>
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/Resources/Strings/CalendarCommonStrings.resx:124]: can you add these for other languages as well?

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/CalendarSkillState.cs [hunk 5]
```diff
@@ -126,6 +127,8 @@ public CalendarSkillState()
 
         public string AskParameterContent { get; set; }
 
+        public string RecursiveToken { get; set; }
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/CalendarSkillState.cs]: I think it's better to name it RecurrencePattern or RecurrenceFrequency

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/UpdateEvent/UpdateEventDialog.cs (hunk 2 above)
Comment (zidaneymar) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/UpdateEvent/UpdateEventDialog.cs]: the event id will be kept as origin.Id.
`updateEvent.Id = origin.Id;`

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/Resources/Strings/CalendarCommonStrings.resx (hunk 4 above)
Comment (zidaneymar) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/Resources/Strings/CalendarCommonStrings.resx:124]: Fixed

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/CalendarSkillState.cs (hunk 5 above)
Comment (zidaneymar) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/CalendarSkillState.cs]: Fixed

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/CalendarSkillDialog.cs (hunk 3 above)
Comment (zidaneymar) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/CalendarSkillDialog.cs]: Fixed

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/UpdateEvent/UpdateEventDialog.cs (hunk 1 above)
Comment (zidaneymar) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/UpdateEvent/UpdateEventDialog.cs]: Fixed.

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/Resources/Strings/CalendarCommonStrings.zh.resx [hunk 6]
```diff
@@ -120,4 +120,16 @@
...
+    <value>每天</value>
+  </data>
+  <data name="MonthlyToken" xml:space="preserve">
+    <value>每月</value>
+  </data>
+  <data name="WeeklyToken" xml:space="preserve">
+    <value>每周</value>
+  </data>
+  <data name="WithTheSubject" xml:space="preserve">
+    <value>以 {0} 为标题</value>
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/Resources/Strings/CalendarCommonStrings.zh.resx]: you don't need the extra spaces around {0}

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/Resources/Strings/CalendarCommonStrings.zh.resx (hunk 6 above)
Comment (zidaneymar) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/Shared/Resources/Strings/CalendarCommonStrings.zh.resx]: Fixed

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**